alot/commands/envelope.py

```diff
@@ -168,6 +168,33 @@ def __init__(self, mail=None, envelope=None, **kwargs):
         self.envelope = envelope
         self.envelope_buffer = None
 
+    def _check_recipient_keys(self):
+        # get all recipient addresses
+        recipients = set()
+        for header in ['To', 'Cc', 'Bcc']:
+            values = self.envelope.headers.get(header)
+            if not values:
+                continue
+            for value in values:
+                for recipient in value.split(','):
+                    address = email.utils.parseaddr(recipient)[1]
+                    recipients.add(address)
+
+        if not recipients:
+            return True
+
+        # get all email addresses of encryption keys
+        keys = set()
+        for key in self.envelope.encrypt_keys.values():
+            for uid in key.uids:
+                keys.add(uid.email)
+
+        # fail if both sets are not equal
```

Comment:
How about computing the key-addresses in a separate helper to simplify the cognitive complexity?